cns/kubecontroller/pod/reconciler.go

```diff
@@ -80,6 +81,54 @@ func (p *watcher) NewNotifierFunc(listOpts *client.ListOptions, limiter limiter,
 	}
 }
 
+// NewPodIPDemandNotifierFunc returns a reconcile.Func optimized for IP demand calculation
+// that uses server-side filtering to exclude terminal Pods (Succeeded/Failed) from the results.
+// This avoids client-side iteration over all pods by using multiple targeted queries.
+func (p *watcher) NewPodIPDemandNotifierFunc(baseListOpts *client.ListOptions, limiter limiter, listeners ...func([]v1.Pod)) reconcile.Func {
```

Comment:
keep the indexer func and setup, but remove this func and revert to the previous filtering strategy in the adapter.go